docs/current_docs/agents/quickstart.mdx

```diff
@@ -0,0 +1,151 @@
+---
+slug: /ai-agents/quickstart
+---
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Build your first coding agent
+
+Estimated time: 5 minutes
+
+In this guide you will create your first coding agent!
+
+The agent will use an LLM of your choosing to solve a coding prompt. The agent will have access to tools to write code and validate it.
+
+## Installation and configuration
+
+Follow the [Dagger AI Agents](/ai-agents#initial-setup) instructions to install and configure Dagger.
+
+## Create a Dagger module
+
+### In an empty directory, create a new Dagger module
+
+First, you will create a Dagger module. This module will be the place where you write your agent functions.
+
+You can write your module in Go, Python, Typescript, PHP, or Java. Choose the SDK you are most comfortable with.
+
+```bash
+dagger init --sdk {your sdk choice} --name coding-agent
+```
+
+This will generate a bunch of files that make up the Dagger module, including some boilerplate functions as examples.
+
+To see the generated functions, run:
+
+```bash
+dagger functions
+```
+
+You should see information about the generated `container-echo` and `grep-dir` functions.
+
+### Install a module dependency
+
+Before you write the agent's function, this agent has a dependency that needs to be installed. This dependency provides the environment the LLM you use to solve the coding prompt.
+
+For more information about the environment concept, see [what is a workspace](/ai-agents#what-is-this-workspace-object-that-most-of-the-examples-use).
+
+```bash
+dagger install github.com/shykes/toy-programmer/toy-workspace
```

Comment:
Shouldn't we create this workspace as part of the quickstart?

Reply:
No, we specifically want to keep this simple and fast. I think explaining the workspace is enough. If we want the user to create the workspace, we're now asking them to create 2 modules and invest 2x the amount of time as their first "try out dagger" experience